lib/vector-core/src/stream/partitioned_batcher.rs

```diff
@@ -287,18 +183,16 @@ impl BatcherSettings {
 }
 
 #[pin_project]
-pub struct PartitionedBatcher<St, Prt, KT>
+pub struct PartitionedBatcher<St, Prt, KT, C, F>
 where
     Prt: Partitioner,
 {
-    /// The total number of bytes a single batch in this struct is allowed to
-    /// hold.
-    batch_allocation_limit: usize,
-    /// The maximum number of items that are allowed per-batch
-    batch_item_limit: usize,
+    /// A closure that retrieves a new [`BatchConfig`] when needed to batch a
+    /// new partition.
+    state: F,
```

Comment:
All users of this batcher appear to pass in a state closure that returns the same thing every time. Am I misreading the callers? Could we instead store that thing with a `Clone` bound so as to just make copies of it?

Reply:
I was thinking to clone an empty batch, which wouldn't have any data yet. However, what you describe makes sense. I don't have strong feelings otherwise, no.